elements/rh-dialog/rh-dialog.ts

```diff
@@ -235,35 +239,126 @@ export class RhDialog extends LitElement {
     }
   }
 
-  @bound private onTriggerClick(event: MouseEvent) {
+  @bound private async onTriggerClick(event: MouseEvent) {
     event.preventDefault();
     this.showModal();
+    await this.updateComplete;
+    this.closeButton?.focus();
   }
 
   @bound private onClick(event: MouseEvent) {
-    const { open, overlay, dialog } = this;
+    const { open, content } = this;
     if (open) {
       const path = event.composedPath();
       const { closeOnOutsideClick } = this.constructor as typeof RhDialog;
 
-      if (closeOnOutsideClick && path.includes(overlay!) && !path.includes(dialog!)) {
+      if (closeOnOutsideClick && !path.includes(content!)) {
         event.preventDefault();
         this.cancel();
       }
     }
   }
 
-  @bound private onKeydown(event: KeyboardEvent) {
+  #trapFocus() {
```

Comment:
why can't we do this with `.showModal()`?

Reply:
Thanks for that feedback! Indeed, those links you referenced are helpful and differ from what the APG says.

I was actually looking at the APG Guidelines for [Dialog (modal)](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/):

> Tab: If focus is on the last tabbable element inside the dialog, moves focus to the first tabbable element inside the dialog.
Shift + Tab: If focus is on the first tabbable element inside the dialog, moves focus to the last tabbable element inside the dialog.

That said, @hellogreg pointed me to a [massively helpful discussion](https://github.com/whatwg/html/issues/8339#issuecomment-1263476659) on this very topic.

tl;dr the way the browser has implemented "focus trapping" with a native HTML `<dialog>` element is indeed correct. The APG guidelines are intended for people not using a native `<dialog>` element.

So, after making the changes Marionne requested, I will update/remove this `#trapFocus()` code and resolve this conversation. 👏